storage/innobase/row/row0upd.cc

```diff
@@ -635,17 +636,26 @@ void row_upd_index_write_log(dict_index_t *index, const upd_t *update,
 
     len = dfield_get_len(new_val);
 
-    /* DD tables do not have their field physical positions determined and they
-    never undergo INSTANT ADD/DROP COLUMN. So use field_no for them or for any
-    indexes that do not have row_versions */
-    ulint field_off = index->has_row_versions() ? upd_field->field_phy_pos
-                                                : upd_field->field_no;
+    ulint field_phy_pos =
+        index->get_field(upd_field->field_no)->col->get_col_phy_pos();
+
+#ifdef UNIV_DEBUG
+    /* Check if multiple fields are updated, they should be different fields
+    in row */
+    if (j > 0) {
```

Comment:
Do we need `j` here ?
What if we reach continue at `if (upd_fld_is_virtual_col(upd_field) && !dict_index_has_virtual(index))` whouldn't `j` be pointing to the wrong filed?

Perhaps we can use `i` instead

Reply:
Other question. The assertion is validating if my current phy_pos is different from previous field pos. 
For example on below table:
```
ID INT pk
col2 INT
col3 INT
col4 INT
``` 

This will have [physical][logical]:
``` 
ID                       [0][0]
DB_TRX                   [1][1]
DB_ROLL_PTR              [2][2]
col2                     [3][3]
col3                     [4][4]
col4                     [5][5]
```
After a DROP col2 INSTANT
```
ID                       [0][0]
DB_TRX                   [1][1]
DB_ROLL_PTR              [2][2]
col3                     [4][3]
col4                     [5][4]
col2                     [3][5]
```

What we were seeing before at crash recover was col3 and col2 having the same phy_pos, but they are 2 logical positions apart now. So I don't this that comparing only with prev field will be enough here.